ideascube/serveradmin/catalog.py

```diff
@@ -762,21 +763,26 @@ def update_cache(self):
 
         for remote in self._remotes.values():
             # TODO: Get resumable.urlretrieve to accept a file-like object?
-            fd, tmppath = tempfile.mkstemp()
-            os.close(fd)
+            with tempfile.NamedTemporaryFile() as fd:
+                tmppath = fd.name
 
-            def _progress(*args):
-                self._progress(' {}'.format(remote.name), *args)
+                def _progress(*args):
+                    self._progress(' {}'.format(remote.name), *args)
 
-            # TODO: Verify the download with sha256sum? Crypto signature?
-            urlretrieve(remote.url, tmppath, reporthook=_progress)
-
-            catalog = load_from_file(tmppath)
-
-            # TODO: Handle content which was removed from the remote source
-            self._available.update(catalog['all'])
-
-            os.unlink(tmppath)
```

Comment:
Hmm, seems not since the `NamedTemporaryFile` gets destroyed as soon as it's closed.